sklearn/feature_selection/_univariate_selection.py

```diff
@@ -297,15 +297,26 @@ def f_regression(X, y, *, center=True):
     else:
         X_norms = row_norms(X.T)
 
-    # compute the correlation
-    corr = safe_sparse_dot(y, X)
-    corr /= X_norms
-    corr /= np.linalg.norm(y)
+    # Compute the correlation
+    with np.errstate(divide="ignore", invalid="ignore"):
+        corr = safe_sparse_dot(y, X)
+        corr /= X_norms
+        corr /= np.linalg.norm(y)
 
-    # convert to p-value
     degrees_of_freedom = y.size - (2 if center else 1)
-    F = corr ** 2 / (1 - corr ** 2) * degrees_of_freedom
-    pv = stats.f.sf(F, 1, degrees_of_freedom)
+
+    # Convert to F-score and then to p-value
+    with np.errstate(divide="ignore", invalid="ignore"):
+        F = corr ** 2 / (1 - corr ** 2) * degrees_of_freedom
+        pv = stats.f.sf(F, 1, degrees_of_freedom)
+
+    # The Pearson correlation coefficient is undefined for random
+    # variables (attributes and target) with a standard deviation
+    # of zero
+    nan_mask = np.isnan(F)
+    F[nan_mask] = 0.0
+    pv[nan_mask] = 1.0
```

Comment:
Lets see what everyone else things of this. We may end up just leaving this as nan.

Aside: If we are going down this path, should the pvalue be 1 or 0?

Reply:
If we decide to replace `F` from `np.nan` to `0.0`, I think that `pv` should be `1.0` for being statistically correct.

For a perfect (negative or positive) correlation (1.0), the p-value is 0.0 (with a (standard) significance level of 0.05, we reject the null hypotheses in favour of the alternative). On the other hand, for no correlation (0.0), the p-value is 1.0 (with a (standard) significance level of 0.05, we cannot reject the null hypotheses in favour of the alternative).

In fact, it makes sense now to me having `np.nan` by `0.0` in `F` and then `pv` equal to `1.0`.

WDYT?

Reply:
Yes I agree that it makes sense that the pvalue is 1.0 for F = 0.